org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/editor/MavenTargetLocationWizard.java

```diff
@@ -95,14 +93,63 @@ public String getText(Object element) {
 					type.setText(targetLocation.getArtifactType());
 					scope.setText(targetLocation.getDependencyScope());
 					metadata.setSelection(new StructuredSelection(targetLocation.getMetadataMode()));
+					bndInstructions = targetLocation.getInstructions(null);
 				} else {
 					artifactId.setText("");
 					groupId.setText("");
 					version.setText("");
 					type.setText(MavenTargetLocation.DEFAULT_PACKAGE_TYPE);
 					scope.setText(MavenTargetLocation.DEFAULT_DEPENDENCY_SCOPE);
 					metadata.setSelection(new StructuredSelection(MavenTargetLocation.DEFAULT_METADATA_MODE));
+					bndInstructions = new BNDInstructions("", null);
 				}
+				updateButton();
+			}
+
+			private void createMetadataCombo(Composite parent) {
+				Composite composite = new Composite(parent, SWT.NONE);
+				composite.setLayout(new GridLayout(2, false));
+				metadata = new ComboViewer(composite);
+				metadata.setContentProvider(ArrayContentProvider.getInstance());
+				metadata.setLabelProvider(new ColumnLabelProvider() {
+					@Override
+					public String getText(Object element) {
+						if (element instanceof MissingMetadataMode) {
+							return ((MissingMetadataMode) element).name().toLowerCase();
+						}
+						return super.getText(element);
+					}
+
+				});
+				editInstructionsButton = new Button(composite, SWT.PUSH);
+				editInstructionsButton.setText("edit manifest generation instructions");
```

Comment:
Ideally, strings should be externalized with NLS.
Also, labels often need to start with an upper case letter; and have a mnemonic on 1 character to enable more keyboard accessibility.

Reply:
![grafik](https://user-images.githubusercontent.com/1331477/99536951-52c94280-29ab-11eb-96e8-4629f699fec1.png)
What do you think? It seems mnemonic can only be used in the text outside the clickable area and only focus it so it looks a bit odd to me.

Maybe just "Edit Instructions" without mnemonic?

Reply:
![grafik](https://user-images.githubusercontent.com/1331477/99537428-f3b7fd80-29ab-11eb-9ced-097825860271.png)

At least one can use tab to activate the link even without mnemonic.